ERCS/erc-7699.md

```diff
@@ -0,0 +1,212 @@
+            // Effectively extract first 32 bytes from transferReference calldata bytes
+            // Note: This is the example. Derivation of the loggedReference is fully up to the implementation.
+            // E.g. keccak hash of the whole transferReference, etc.
+            bytes32 loggedReference;
+            // solhint-disable-next-line no-inline-assembly
+            assembly {
+                loggedReference := calldataload(transferReference.offset)
+            }
+
+            emit TransferReference(loggedReference);
+        }
+    }
+
+    /**
+     * @notice A standard ERC20 token transfer with an optional transfer reference
+     * @dev The underlying `transfer` function is assumed to handle the actual token transfer logic.
+     * @param to The address of the recipient where the tokens will be sent.
+     * @param amount The number of tokens to be transferred.
+     * @param transferReference A bytes field to include a transfer reference, reference signature or other relevant
+     * reference data.
+     * @return A boolean indicating whether the transfer was successful.
+     */
+    function transfer(address to, uint256 amount, bytes calldata transferReference) public virtual returns (bool) {
+        _logReference(transferReference);
+
+        // ERC20.Transfer event is emitted immediately after TransferReference event
+        return transfer(to, amount);
+    }
+
+    /**
+     * @notice A delegated token transfer with an optional transfer reference
+     * @dev Requires prior approval from the token owner. The underlying `transferFrom` function is assumed to handle
+     * allowance and transfer logic.
+     * @param from The address of the token owner who has authorized the transfer.
+     * @param to The address of the recipient where the tokens will be sent.
+     * @param amount The number of tokens to be transferred.
+     * @param transferReference A bytes field to include a transfer reference, reference signature or other relevant
+     * reference data.
+     * @return A boolean indicating whether the transfer was successful.
+     */
+    function transferFrom(address from, address to, uint256 amount, bytes calldata transferReference)
+        public
+        virtual
+        returns (bool)
+    {
+        _logReference(transferReference);
+
+        // ERC20.Transfer event is emitted immediately after TransferReference event
+        return transferFrom(from, to, amount);
+    }
+}
+
+```
+
+## Security Considerations
+
+### Privacy Considerations
+
+Reference data privacy: Including payment references in token transfers may expose sensitive information about the transaction or the parties involved. Implementers and users SHOULD carefully consider the privacy implications and ensure that payment references do not reveal sensitive information. To mitigate this risk, implementers can consider using encryption or other privacy-enhancing techniques to protect payment reference data.
```

Comment:
```suggestion
Reference data privacy: Including payment references in token transfers may expose sensitive information about the transaction or the parties involved. Implementers and users should carefully consider the privacy implications and ensure that payment references do not reveal sensitive information. To mitigate this risk, implementers can consider using encryption or other privacy-enhancing techniques to protect payment reference data.
```

Avoid using the keywords outside the specification section.

Reply:
fixed with #716  Thx